### PR TITLE
Add i18n

### PR DESCRIPTION
**Pull request by tavriaforever**
// cc @gela-d @tadatuta @tormozz48

### CODE REVIEW

File: desktop.blocks/forum-date/forum-date.bemhtml [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
+                3: BEM.I18N('forum-date', 'apr'),
+                4: BEM.I18N('forum-date', 'may'),
+                5: BEM.I18N('forum-date', 'jun'),
+                6: BEM.I18N('forum-date', 'jul'),
+                7: BEM.I18N('forum-date', 'aug'),
+                8: BEM.I18N('forum-date', 'sep'),
+                9: BEM.I18N('forum-date', 'oct'),
+                10: BEM.I18N('forum-date', 'nov'),
+                11: BEM.I18N('forum-date', 'dec')
+            };
```
Comment (sipayRT) [inline desktop.blocks/forum-date/forum-date.bemhtml]: почему не обычный массив?

Comment (tadatuta) [inline desktop.blocks/forum-date/forum-date.bemhtml]: yeah, array would be better. and `var i18n = BEM.I18N`

File: desktop.blocks/forum-date/forum-date.i18n/ru.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+module.exports = {
+    "forum-date": {
+        "jan": "Января",
```
Comment (tadatuta) [inline desktop.blocks/forum-date/forum-date.i18n/ru.js]: why capitalised?

File: desktop.blocks/issue/__solve/issue__solve.bemhtml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    js()(true),
+
+    content()(function() {
+        var ctx = this.ctx;
+
+        return {
+            block: 'link',
+            mods: { pseudo : true },
+            title: ctx.isClosed ? BEM.I18N('issue', 'state-set-open') : BEM.I18N('issue', 'state-set-close'),
```
Comment (tadatuta) [inline desktop.blocks/issue/__solve/issue__solve.bemhtml]: `title: BEM.I18N('issue', ctx.isClosed ? 'state-set-open' : 'state-set-close'),`

File: desktop.blocks/issue/__solve/issue__solve.js [hunk 4]
```diff
@@ -36,9 +36,11 @@ modules.define('issue', ['jquery'], function(provide, $, Issue) {
          * @private
          */
         _onClickSolve: function() {
-            var isSolved = this.hasMod('solved');
+            var isSolved = this.hasMod('solved'),
+                i18n = this.elemParams('solve').i18n,
+                message = (isSolved ? i18n['open-message'] : i18n['close-message']) + ' ' + i18n['post'].toLowerCase() + ' ?';
```
Comment (tadatuta) [inline desktop.blocks/issue/__solve/issue__solve.js]: `message = [i18n[isSolved ? 'open-message' : 'close-message'], i18n.post.toLowerCase(), '?'].join(' ');`

File: server/template.js [hunk 5]
```diff
@@ -8,21 +8,11 @@ var u = require('util'),
     stringify = require('json-stringify-safe'),
```
Comment (tadatuta) [inline server/template.js:8]: take a look at BEMINFO-647, we've got the same thing here :(

File: desktop.blocks/forum-date/forum-date.i18n/ru.js (hunk 2 above)
Comment (tavriaforever) [inline desktop.blocks/forum-date/forum-date.i18n/ru.js]: I don`t understand question

File: desktop.blocks/issue/__solve/issue__solve.js (hunk 4 above)
Comment (tavriaforever) [inline desktop.blocks/issue/__solve/issue__solve.js]: for me it's not readable

**Comment by tavriaforever** [comment]
update

File: desktop.blocks/forum-date/forum-date.i18n/ru.js (hunk 2 above)
Comment (sipayRT) [inline desktop.blocks/forum-date/forum-date.i18n/ru.js]: Why Do You Use Uppercase Letters Here? :)

**Comment by tavriaforever** [comment]
update 2

File: bower.json [hunk 6]
```diff
@@ -29,6 +29,7 @@
   },
   "resolutions": {
     "bem-core": "v2.5.0",
-    "bem-components": "b2e0a63522f0636da40996eb6d4042c09f48e1ff"
+    "bem-components": "b2e0a63522f0636da40996eb6d4042c09f48e1ff",
+    "bem-social": "master"
```
Comment (tadatuta) [inline bower.json]: kinda palevo. please fix on particular hash if there's no tag released

File: desktop.blocks/forum-date/forum-date.i18n/en.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+module.exports = {
+    "forum-date": {
+        "jan": "january",
```
Comment (tadatuta) [inline desktop.blocks/forum-date/forum-date.i18n/en.js]: actually in English names of the months start with a capital letter...

Comment (sipayRT) [inline desktop.blocks/forum-date/forum-date.i18n/en.js]: eee!!!

File: desktop.blocks/forum-header/__title/forum-header__title.styl [hunk 8]
```diff
@@ -1,4 +1,4 @@
-$icon-logo-size = 10px;
+$icon-logo-size = 9px;
```
Comment (sipayRT) [inline desktop.blocks/forum-header/__title/forum-header__title.styl:1]: not sure that these changes are about i18n

File: desktop.blocks/forum-header/_view/forum-header_view_auth.bemtree [hunk 9]
```diff
@@ -45,7 +45,7 @@ block('forum-header').mod('view', 'auth')(
         def()(function() {
             return {
                 block : 'button',
-                text : 'Добавить пост',
+                text : BEM.I18N('forum-header', 'add-button'),
```
Comment (sipayRT) [inline desktop.blocks/forum-header/_view/forum-header_view_auth.bemtree]: so what codestyle about spase before `:` you will use?

File: desktop.blocks/forum-header/forum-header.i18n/en.js [hunk 10]
```diff
@@ -0,0 +1,10 @@
+module.exports = {
+    "forum-header": {
+        "logo": "Forum",
+        "status": "Beta",
+        "bug-button-title": "Report bugs",
+        "github-button-title": "Repository forum",
+        "add-button": "Submit new post",
```
Comment (sipayRT) [inline desktop.blocks/forum-header/forum-header.i18n/en.js]: add-button !== submit new post.
maybe at least `add-post-button`?

File: desktop.blocks/forum-sorting/forum-sorting.i18n/ru.js [hunk 11]
```diff
@@ -0,0 +1,10 @@
+module.exports = {
+    "forum-sorting": {
+        "sort-newest": "Новые",
+        "sort-oldest": "Старые",
+        "sort-most-commented": "Наиболее комментируемые",
+        "sort-least-commented": "Наименее комментируемые",
+        "sort-updated": "Последние обновленные",
+        "sort-least-updated": "Давно обновленные"
```
Comment (sipayRT) [inline desktop.blocks/forum-sorting/forum-sorting.i18n/ru.js:8]: what does it mean?

File: desktop.blocks/forum/forum.i18n/ru.js [hunk 12]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+    "forum": {
+        "title-placeholder": "Заголовок",
+        "body-placeholder": "Описание",
+        "title-labels": "Метки",
+        "button-add": "Добавить",
+        "button-cancel": "Отменить"
```
Comment (sipayRT) [inline desktop.blocks/forum/forum.i18n/ru.js]: you can change button to link and this name will be wrong. So you can change these names to `add` and `cancel`. i18n just about it

File: desktop.blocks/issue/issue.i18n/en.js [hunk 13]
```diff
@@ -0,0 +1,19 @@
+module.exports = {
+    "issue": {
+        "state-set-open": "Open",
+        "state-set-close": "Close",
+        "edit-form-title-placeholder": "Title",
```
Comment (sipayRT) [inline desktop.blocks/issue/issue.i18n/en.js:5]: the same point about naming of keys

File: desktop.blocks/forum-form/forum-form.js [hunk 14]
```diff
@@ -164,22 +160,22 @@ modules.define('forum-form', ['i-bem__dom', 'jquery'], function(provide, BEMDOM,
...
-                    return (!checkboxByName.hasMod('checked', true));
-                });
-
-            if(checked) {
-                this._showError('notChecked', errorText ? errorText : false);
-
-                return true;
+        isEmptyCheckbox: function(name) {
+            if(this.params.labelsRequired) {
+                var checked = this.findBlocksInside(this.elem('control'), 'checkbox')
```
Comment (tadatuta) [inline desktop.blocks/forum-form/forum-form.js:167]: I'm not quite happy with all this findBlocksInside().filter().every() to find control by name.
Maybe you can plan refactoring and mix `forum-form__{name}` to each so it'd be just `this.findBlockInside(name, 'checkbox').hasMod('checked')`?

Comment (tadatuta) [inline desktop.blocks/forum-form/forum-form.js:167]: everything else looks fine

File: desktop.blocks/forum-sorting/forum-sorting.i18n/ru.js (hunk 11 above)
Comment (tavriaforever) [inline desktop.blocks/forum-sorting/forum-sorting.i18n/ru.js:8]: your variant?

File: desktop.blocks/forum-header/_view/forum-header_view_auth.bemtree (hunk 9 above)
Comment (tavriaforever) [inline desktop.blocks/forum-header/_view/forum-header_view_auth.bemtree]: ok

File: desktop.blocks/forum-header/forum-header.i18n/en.js (hunk 10 above)
Comment (tavriaforever) [inline desktop.blocks/forum-header/forum-header.i18n/en.js]: agree

File: desktop.blocks/forum-date/forum-date.i18n/en.js (hunk 7 above)
Comment (tavriaforever) [inline desktop.blocks/forum-date/forum-date.i18n/en.js]: :angry:

File: desktop.blocks/forum/forum.i18n/ru.js (hunk 12 above)
Comment (tavriaforever) [inline desktop.blocks/forum/forum.i18n/ru.js]: ok

File: desktop.blocks/forum-form/forum-form.js (hunk 14 above)
Comment (tavriaforever) [inline desktop.blocks/forum-form/forum-form.js:167]: ok

File: desktop.blocks/forum-sorting/forum-sorting.i18n/ru.js (hunk 11 above)
Comment (sipayRT) [inline desktop.blocks/forum-sorting/forum-sorting.i18n/ru.js:8]: I don't know what it means

File: desktop.blocks/forum-header/__title/forum-header__title.styl (hunk 8 above)
Comment (tavriaforever) [inline desktop.blocks/forum-header/__title/forum-header__title.styl:1]: Add one more commit about this changes

**Comment by tavriaforever** [comment]
rebased, ready to merge

File: desktop.blocks/forum-sorting/forum-sorting.i18n/ru.js (hunk 11 above)
Comment (tavriaforever) [inline desktop.blocks/forum-sorting/forum-sorting.i18n/ru.js:8]: See github issues sorting - one of the variants is `least recently updated`

**Pull request merged**